init_features.go

```diff
@@ -310,6 +310,10 @@ func buildNamespaceMemo(memo internal.NamespaceMemo, typ reflect.Type) error {
 	} else if typ.Kind() == reflect.Struct && typ != reflect.TypeOf(time.Time{}) {
 		structName := typ.Name()
 		namespace := internal.ChalkpySnakeCase(structName)
+		if _, ok := memo[structName]; ok {
+			// Prevent infinite loops
+			return nil
+		}
```

Comment:
why is this recursion necessary only in this case? the first case sure looks infinite loopy to me

Reply:
first case as in the pointer case? I think that recurses in a definite direction, as in from the pointer to a type it recurses into the pointed to type. The only case where we jump to another namespace/struct is within another struct.

Reply:
Basically 'typ.Elem()' means the type that 'typ' points to